monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,307 @@
+        self.sobel_gradient = SobelGradients(kernel_size=kernel_size)
+        if remove_small_objects:
+            self.remove_small_objects = RemoveSmallObjects(min_size=min_size)
+        else:
+            self.remove_small_objects = None  # type: ignore
+
+    def __call__(self, mask: NdarrayOrTensor, hover_map: NdarrayOrTensor) -> NdarrayOrTensor:  # type: ignore
+        """
+        Args:
+            mask: binarized segmentation result.  Shape must be [1, H, W].
+            hover_map:  horizontal and vertical distances of nuclear pixels to their centres of mass. Shape must be [2, H, W].
+                The first and second channel represent the horizontal and vertical maps respectively. For more details refer
+                to papers: https://arxiv.org/abs/1812.06499.
+
+        Return:
+            Foreground probability map.
+        """
+        if len(mask.shape) != 3 or len(hover_map.shape) != 3:
+            raise ValueError(
+                f"Suppose the mask and hover map should be with shape of [C, H, W], but got {mask.shape}, {hover_map.shape}"
+            )
+        if mask.shape[0] != 1:
+            raise ValueError(f"Suppose the mask only has one channel, but got {mask.shape[0]}")
+        if hover_map.shape[0] != 2:
+            raise ValueError(f"Suppose the hover map only has two channels, but got {hover_map.shape[0]}")
+
+        hover_h = hover_map[0:1, ...]
+        hover_v = hover_map[1:2, ...]
+
+        hover_h = (hover_h - min(hover_h)) / (max(hover_h) - min(hover_h))
+        hover_v = (hover_v - min(hover_v)) / (max(hover_v) - min(hover_v))
+        sobelh = self.sobel_gradient(hover_h)[0, ...]
+        sobelv = self.sobel_gradient(hover_v)[1, ...]
+        sobelh = 1 - (sobelh - min(sobelh)) / (max(sobelh) - min(sobelh))
+        sobelv = 1 - (sobelv - min(sobelv)) / (max(sobelv) - min(sobelv))
+
+        # combine the h & v values using max
+        overall = maximum(sobelh, sobelv)
+        overall = overall - (1 - mask)
+        overall[overall < 0] = 0
+
+        return convert_to_dst_type(overall, mask, dtype=np.float32)[0]
+
+
+class GenerateDistanceMap(Transform):
+    """
+    Generate distance map.
+    In general, the instance map is calculated from the distance to the background.
+    Here, we use 1 - "foreground probability map" to generate the distance map.
+    Nuclei values form mountains so inverse to get basins.
+
+    Args:
+        smooth_fn: execute smooth function on distance map. Defaults to None. You can specify
+            callable functions for smoothing.
+            For example, if you want apply gaussian smooth, you can specify `smooth_fn = GaussianSmooth()`
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(self, smooth_fn: Optional[Callable] = None) -> None:
+        self.smooth_fn = smooth_fn
+
+    def __call__(self, mask: NdarrayOrTensor, foreground_prob_map: NdarrayOrTensor) -> NdarrayOrTensor:  # type: ignore
```

Comment:
Is the input argument foreground_prob_map a general concept beyond this work? In other words, can it be calculate in any other way than using `GenerateProbabilityMap`?